File: python/fachada/HomeTheaterFacade.py
# Facade Class
class HomeTheaterFacade:
    def __init__(self, amplifier, dvd_player, projector):
        self.amplifier = amplifier
        self.dvd_player = dvd_player
        self.projector = projector

    def watch_movie(self):
        print("Setting up the home theater...")
        self.amplifier.on()
        self.projector.on()
        self.dvd_player.play()
        print("Movie is ready to watch!")

    def end_movie(self):
        print("Shutting down the home theater...")
        self.dvd_player.stop()
        self.projector.off()
        self.amplifier.off()
        print("Home theater is off.")

File: java/fachada/fachada/src/main/java/edu/uea/HomeTheaterFacade.java
package edu.uea;

// Facade Class
class HomeTheaterFacade {
    private Amplifier amplifier;
    private DVDPlayer dvdPlayer;
    private Projector projector;

    public HomeTheaterFacade(Amplifier amplifier, DVDPlayer dvdPlayer, Projector projector) {
        this.amplifier = amplifier;
        this.dvdPlayer = dvdPlayer;
        this.projector = projector;
    }

    public void watchMovie() {
        System.out.println("Setting up the home theater...");
        amplifier.on();
        projector.on();
        dvdPlayer.play();
        System.out.println("Movie is ready to watch!");
    }

    public void endMovie() {
        System.out.println("Shutting down the home theater...");
        dvdPlayer.stop();
        projector.off();
        amplifier.off();
        System.out.println("Home theater is off.");
    }
}